skbio/diversity/alpha/_base.py

```diff
@@ -848,6 +850,51 @@ def osd(counts):
     return observed_otus(counts), singles(counts), doubles(counts)
 
 
+@experimental(as_of="0.4.0-dev")
+def pielou_e(counts):
+    r"""Calculate Pielou's Evenness index J'.
+
+    Pielou's Evenness is defined as:
+
+    .. math::
+
+       J' = \frac{(H)}{\ln(S)}
+
+    where :math:`H` is the Shannon-Wiener entropy of counts and :math:`S` is
+    the number of OTUs in the sample.
+
+    Parameters
+    ----------
+    counts : 1-D array_like, int
+        Vector of counts.
+
+    Returns
+    -------
+    double
+        Pielou's Evenness.
+
+    See Also
+    --------
+    shannon
+    heip_e
+
+    Notes
+    -----
+    The implementation here is based on the description given by Carlo Heip
+    [1]_ and is also described on Wikipedia [2]_. It is similar to Heip's
+    evenness [1]_.
+
+    References
+    ----------
+    .. [1] Heip, C. 1974. A new index measuring evenness. J. Mar. Biol. Ass.
+       UK., 54, 555-557.
+    .. [2] https://en.wikipedia.org/wiki/Species_evenness
+
+    """
+    counts = _validate_counts_vector(counts)
+    return shannon(counts) / np.log(observed_otus(counts))
```

Comment:
Sounds like a plan @jairideout!

How do I test for `nan`. This does not work:
``` python
FAIL: test_pielou_e (skbio.diversity.alpha.tests.test_base.BaseTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/bris469/Documents/Projects/toy/scikit-bio/skbio/diversity/alpha/tests/test_base.py", line 414, in test_pielou_e
    self.assertEqual(pielou_e([0, 0, 200, 0, 0]), np.nan)
AssertionError: nan != nan
```

Both correctly return the `numpy.nan` object which is a `numpy.float64`. However, `assertEqual` gets scared when it see this.